### ISSUE TITLE
Lamba Memory Size Increment Changes

### ISSUE THREAD

**Issue by MattGirolami** [aws-cloudformation/cfn-python-lint#1857]
*cfn-lint version: (`cfn-lint --version`)*
0.44.2

*Description of issue.*
Lambda now allows arbitrary MemorySize values rather than in increments of 64. Additionally, the max memory size for a Lambda function was increased to 10GB

cfn-lint marks memory sizes that are not in increments of 64 as invalid values

Please provide as much information as possible:
* Template linting issues:
  * Please provide a CloudFormation sample that generated the issue.
  
```yaml
Resources:
  Lambda:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: Function
      Handler: foo.bar.baz.App::handleRequest
      Runtime: java8
      MemorySize: 511
      Timeout: 300
      ReservedConcurrentExecutions: 1
      Policies:
            - arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole
```

  * If present, please add links to the (official) documentation for clarification.
  https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-memorysize
  
  
  https://aws.amazon.com/blogs/aws/new-for-aws-lambda-functions-with-up-to-10-gb-of-memory-and-6-vcpus/
  
  * Validate if the issue still exists with the latest version of `cfn-lint` and/or the latest Spec files
  Present with cfn-lint 0.44.2
* Feature request:
  * Please provide argumentation about the missing feature. Context is key!
  
  Support arbitrary memory megabyte values between 128-10240 

Cfn-lint uses the [CloudFormation Resource Specifications](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html) as the base to do validation. These files are included as part of the application version. Please update to the latest version of `cfn-lint` or update the spec files manually (`cfn-lint -u`)
